kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java

```diff
@@ -155,10 +155,8 @@ private KaldbMetadataStoreChangeListener cacheNodeListener() {
 
   private void registerSearchMetadata(String snapshotName)
       throws ExecutionException, InterruptedException, TimeoutException {
-    // for "name" we pass url so that we can run two JVMs on the same host and not have znode path
-    // collision.
     this.searchMetadata =
-        new SearchMetadata(searchContext.toName(), snapshotName, searchContext.toUrl());
+        new SearchMetadata(snapshotName + "_" + searchContext.hostname, snapshotName, searchContext.toUrl());
```

Comment:
Prevents collisions when a single node needs to register multiple search metadata (ie, cache nodes)